crates/fs/src/fs.rs

```diff
@@ -548,6 +548,10 @@ impl Fs for RealFs {
         let pending_paths: Arc<Mutex<Vec<PathEvent>>> = Default::default();
         let root_path = path.to_path_buf();
 
+        // Check if root path is a symlink
+        let tr_path = self.read_link(&path).await.ok();
+        let tr_path_clone = tr_path.clone();
```

Comment:
Using `.ok` here because the error type returned by `.read_link` doesn't implement `Clone` and we don't actually use the error so an `Option` does the job.